pkg/storage/db.go

```diff
@@ -790,25 +791,29 @@ func (db *DB) runQuery(ctx context.Context, query string, args ...interface{}) (
 }
 
 func (db *DB) assignPrivateIP(ctx context.Context, tx *sql.Tx, arnID string, ip string, when time.Time) error {
-	const assignPrivateIPQuery = `
-do
-$$
-    begin
-        update aws_private_ip_assignment
-        set not_before = $1
-        where private_ip = $2
-          and not_before = to_timestamp(0)
-          and not_after > $1
-          and aws_resource_id = (select id from aws_resource where arn_id = $3);
-        if not found then
-            insert into aws_private_ip_assignment
-                (not_before, private_ip, aws_resource_id)
-            values ($1, $2, (select id from aws_resource where arn_id = $3));
-        end if;
-    end
-$$;
-`
-	_, err := tx.ExecContext(ctx, assignPrivateIPQuery, when, ip, arnID)
+	const assignPrivateIPQueryUpdate = `
+update aws_private_ip_assignment
+set not_before = $1
+where private_ip = $2
+  and not_before = to_timestamp(0)
+  and not_after > $1
+  and aws_resource_id = (select id from aws_resource where arn_id = $3);`
+	const assignPrivateIPQueryInsert = `
+insert into aws_private_ip_assignment
+    (not_before, private_ip, aws_resource_id)
+values ($1, $2, (select id from aws_resource where arn_id = $3)) on conflict do nothing ;`
+	res, err := tx.ExecContext(ctx, assignPrivateIPQueryUpdate, when, ip, arnID)
```

Comment:
nit: a newline here(after the  sql) would help separate the sql strings from the go logic and improve readability